ingestion_server/ingestion_server/indexer.py

```diff
@@ -469,3 +411,60 @@ def point_alias(self, model_name: str, index_suffix: str, alias: str, **_):
         if self.progress is not None:
             self.progress.value = 100  # mark job as completed
         self.ping_callback()
+
+    def delete_index(
+        self,
+        model_name: str,
+        index_suffix: Optional[str] = None,
+        alias: Optional[str] = None,
+        force_delete: bool = False,
+        **_,
+    ):
+        """
+        Delete the given index ensuring that it is not in use.
+
+        :param model_name: the name of the media type
+        :param index_suffix: the suffix of the index to delete
+        :param alias: the alias to delete, including the index it points to
+        :param force_delete: whether to delete the index even if it is in use
+        """
+
+        target = alias if alias is not None else f"{model_name}-{index_suffix}"
+
+        target_stat = get_stat(self.es, target)
+        if target_stat.exists:
+            if target_stat.is_alias:
+                if not force_delete:
+                    # Alias cannot be deleted unless forced.
+                    message = (
+                        f"Alias {target} might be in use so it cannot be deleted. Use"
+                        f"the `force_delete` parameter."
```

Comment:
If we can give more specific instructions for how to check this too (maybe another request or a link to a document), that would be really helpful for making it as easy as possible for folks to confirm this before using force_delete.

```suggestion
                        f"Alias {target} might be in use so it cannot be deleted. Confirm "
                        " that it is not in use, then use the `force_delete` parameter."
```

Reply:
Updated the message in 44e0d0a.